Lib/fontTools/varLib/models.py

```diff
@@ -61,13 +61,16 @@ def normalizeValue(v, triple, extrapolate=False):
         )
     if not extrapolate:
         v = max(min(v, upper), lower)
-    if v == default:
-        v = 0.0
-    elif v < default:
-        v = (v - default) / (default - lower)
-    else:
-        v = (v - default) / (upper - default)
-    return v
+
+    if v == default or lower == upper:
+        return 0.0
+
+    if (v < default and lower != default) or (v > default and upper == default):
+        return (v - default) / (default - lower)
+    elif (v > default and upper != default) or (v < default and lower == default):
+        return (v - default) / (upper - default)
+
+    raise AssertionError("that's not possible")
```

Comment:
I'd go with ValueError.

Reply:
it's deliberately AssertionError because it can't _ever_ be hit if the logic above is correct

Reply:
basically that line is never executed

Reply:
> Then the previous elif can be an else, no?

yes, but I wanted to make the conditions explicit.. I can change to else (and add the conditions in comments)

Reply:
Or assert the condition.

Reply:
@justvanrossum before I make any changes, first we need to agree on the semantics of this code. What do you think of https://github.com/fonttools/fonttools/pull/2849#discussion_r994953694 ?